adcm_aio_client/core/requesters.py

```diff
@@ -22,20 +40,129 @@ def as_dict(self: Self) -> dict: ...
 
 
 class Requester(Protocol):
-    async def get(self: Self, path: str, query_params: dict) -> AsyncGenerator[RequesterResponse]: ...
+    async def get(self: Self, path: str, query_params: dict) -> RequesterResponse: ...
+
+    async def post(self: Self, path: str, data: dict) -> RequesterResponse: ...
+
+    async def patch(self: Self, path: str, data: dict) -> RequesterResponse: ...
+
+    async def delete(self: Self, path: str) -> RequesterResponse: ...
+
+
+@dataclass(slots=True, frozen=True)
+class Credentials:
+    username: str
+    password: str
+
+    def dict(self: Self) -> dict:
+        return asdict(self)
+
+    def __repr__(self: Self) -> str:
+        return f"{self.username}'s credentials"
+
+
+@dataclass(slots=True, frozen=True)
+class HTTPXRequesterResponse:
```

Comment:
let's separate protocols from implementations

@a-alferov how do you see it?

Reply:
Yes, I agree. How about storing all the protocols in a file `types.py`?

Reply:
@a-alferov we may try, but have to revisit it before first release, it may be too big

Reply:
@DanBalalan do it now, pls

Reply:
moved